src/itemset.js

```diff
@@ -203,30 +211,51 @@ define([
     }
 
     function itemsetWidget(mug, options) {
-        var widget = datasources.dataSourceWidget(mug, options, "Data Source"),
+        var widget = datasources.fixtureWidget(mug, options, "Data Source"),
             super_getUIElement = widget.getUIElement,
             super_getValue = widget.getValue,
             super_setValue = widget.setValue,
-            handleChange = widget.handleChange.bind(widget),
-            labelRef = refSelect("label_ref", "Choice Label", widget.isDisabled()),
-            valueRef = refSelect("value_ref", "Choice Value", widget.isDisabled());
+            super_handleChange = widget.handleChange.bind(widget),
+            labelRef = refSelect("label_ref", "Choice Label", false),
+            valueRef = refSelect("value_ref", "Choice Value", false),
+            filterRef = refSelect("filter_ref", "Choice Filter", false);
+
+        function updateAutoComplete() {
+            if (widget.getValue().instance) {
+                var sources = datasources.autocompleteChoices(widget.getValue().instance.src);
+                labelRef.addAutoComplete(sources);
+                valueRef.addAutoComplete(sources);
+            }
+        }
 
-        labelRef.onChange(handleChange);
-        valueRef.onChange(handleChange);
+        var local_handleChange = function() {
+            updateAutoComplete();
+            super_handleChange();
+        };
+
+        widget.handleChange = local_handleChange;
+
+        labelRef.onChange(super_handleChange);
+        valueRef.onChange(super_handleChange);
+        filterRef.onChange(super_handleChange);
 
         widget.getUIElement = function () {
             return super_getUIElement()
                 .append(labelRef.element)
-                .append(valueRef.element);
+                .append(valueRef.element)
+                .append(filterRef.element);
         };
 
         widget.getValue = function () {
-            var val = super_getValue();
+            var val = super_getValue(),
+                s = val.query.search(END_FILTER),
+                filter = s === -1 ? '' : val.query.slice(s);
```

Comment:
Could change this to
```js
filter = END_FILTER.test(val.query) ? '' : val.query.slice(s);
```

Is it ever possible for `val.query` be anything other than a string?